moped-editor/src/utils/mapDrawHelpers.js

```diff
@@ -1,223 +1,15 @@
-  DrawLineStringMode,
-  DrawPointMode,
-  EditingMode,
-  RENDER_STATE,
-  SHAPE,
-} from "react-map-gl-draw";
-import { v4 as uuidv4 } from "uuid";
+import React, { useState, useRef } from "react";
+import ComponentsDrawControl from "src/components/Maps/ComponentsDrawControl";
 import { get } from "lodash";
-import theme from "../theme/index";
-import { mapStyles, drawnLayerNames } from "../utils/mapHelpers";
-
-export const MODES = [
-  {
-    id: "disableDrawMode",
-    text: "Select & Move",
-    handler: null,
-    icon: "icon-pointer.svg",
-  },
-  {
-    id: "drawPoint",
-    text: "Draw Point",
-    handler: DrawPointMode,
-    icon: "icon-draw-marker.svg",
-  },
-  {
-    id: "drawLine",
-    text: "Draw Line",
-    handler: DrawLineStringMode,
-    icon: "icon-draw-lines.svg",
-  },
-  {
-    id: "edit",
-    text: "Select Point",
-    handler: EditingMode,
-    icon: "icon-delete.svg",
-  },
-];
-
-export const POINT_MODES = [
-  {
-    id: "disableDrawMode",
-    text: "Select & Move",
-    handler: null,
-    icon: "icon-pointer.svg",
-  },
-  {
-    id: "drawPoint",
-    text: "Draw Point",
-    handler: DrawPointMode,
-    icon: "icon-draw-marker.svg",
-  },
-  {
-    id: "edit",
-    text: "Select Point",
-    handler: EditingMode,
-    icon: "icon-delete.svg",
-  },
-];
-
-export const LINE_MODES = [
-  {
-    id: "disableDrawMode",
-    text: "Select & Move",
-    handler: null,
-    icon: "icon-pointer.svg",
-  },
-  {
-    id: "drawLine",
-    text: "Draw Line",
-    handler: DrawLineStringMode,
-    icon: "icon-draw-lines.svg",
-  },
-  {
-    id: "edit",
-    text: "Select Point",
-    handler: EditingMode,
-    icon: "icon-delete.svg",
-  },
-];
-
-const STROKE_COLOR = theme.palette.primary.main;
-const FILL_COLOR = theme.palette.primary.main;
-
-const SELECTED_STYLE = {
-  stroke: STROKE_COLOR,
-  strokeWidth: 8,
-  fill: FILL_COLOR,
-  fillOpacity: 0,
-};
-
-const HOVERED_STYLE = {
-  stroke: STROKE_COLOR,
-  strokeWidth: 8,
-  fill: FILL_COLOR,
-  fillOpacity: 0,
-};
-
-const DEFAULT_STYLE = {
-  stroke: theme.palette.primary.main,
-  strokeWidth: 4,
-  fill: theme.palette.secondary.main,
-  fillOpacity: 1,
-};
-
-/**
- * Interpolate a feature width based on the zoom level of map
- * Adapted from Mapbox GL JS linear interpolation using a formula linked below
- * https://github.com/mapbox/mapbox-gl-js/blob/d66ff288e7ab2e917e9e676bee942dd6a46171e7/src/style-spec/expression/definitions/interpolate.js
- * https://matthew-brett.github.io/teaching/linear_interpolation.html
- * @param {number} currentZoom - Current zoom level from the map
- * @param {number} minZoom - Minimum zoom level from the current bracket
- * @param {number} maxZoom - Maximum zoom level from the current bracket
- * @param {number} minPixelWidth - Minimum pixel width from the current bracket
- * @param {number} maxPixelWidth - Maximum pixel width from the current bracket
- * @return {number} Interpolated pixel width
- */
-function linearInterpolation(
```

Comment:
With Mapbox GL Draw, we can now feed the [already existing styles that use zoom-driven stops](https://github.com/cityofaustin/atd-moped/blob/e70a150846198ae5d9cd99a77ebabaee952d2211/moped-editor/src/utils/mapHelpers.js#L78-L95) for circle radius and line width. So, we don't need to do this manually anymore.

Reply:
really nice to see all the red!